### PR TITLE
Export ExtensibleTrustManager service interface

### PR DESCRIPTION
**Pull request by mgbowman**

### CODE REVIEW

File: bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/ExtensibleTrustManager.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.io.net.http;
+
+import javax.net.ssl.TrustManager;
+
+public interface ExtensibleTrustManager extends TrustManager {
```
Comment (martinvw) [inline bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/ExtensibleTrustManager.java]: This class will need some class level docblock

Comment (mgbowman) [reply bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/ExtensibleTrustManager.java]: I didn't know if it was going to be ok for me to commit this to you - I don't have an ESH developer agreement.

I will update this now

**Comment by mgbowman** [comment]
Didn't know if I need any --signoff as this is going to your repo which will go upsteam to ESH. I just made this to make it more clear what I was talking about in our thread.

**Comment by mgbowman** [comment]
> This was indeed missing, is this enough to inject it and add providers in the fly? Thanks

Yes, quick testing injects the ETM and I added a `host:port` based `TlsTrustManagerProvider` and it worked as we expected!

**Pull request merged**